java/vector/src/main/java/org/apache/arrow/vector/BaseNullableVariableWidthVector.java

```diff
@@ -836,17 +858,259 @@ public int getLastSet() {
     * @return starting offset for the element
     */
    public long getStartEnd(int index) {
-      return (long)offsetBuffer.getInt(index * OFFSET_WIDTH);
+      return offsetBuffer.getLong(index * OFFSET_WIDTH);
    }
 
    /**
     * Mark the particular position in the vector as non-null.
     *
     * @param index position of the element.
     */
+   @Override
    public void setIndexDefined(int index) {
-      handleSafe(index, 0);
+      while (index >= getValidityBufferValueCapacity()) {
+         validityBuffer = reallocBufferHelper(validityBuffer, false);
+      }
+      BitVectorHelper.setValidityBitToOne(validityBuffer, index);
+   }
+
+   /**
+    * Sets the value length for an element.
+    *
+    * @param index   position of the element to set
+    * @param length  length of the element
+    */
+   public void setValueLengthSafe(int index, int length) {
```

Comment:
I am curious what's the use case of this API, i.e, when do we want to just change the length?

Reply:
@siddharthteotia what's your thought on this?

Reply:
Ping @siddharthteotia , can you explain why do we need this as a public API? When would a user want to just sent length for an element?